### PR TITLE
Feature/seext 4533

### PR DESCRIPTION
**Pull request by SoHotSoup**

### CODE REVIEW

**Comment by ikatun** [comment]
This is the error I get when I try to run the platform from this branch:

```
C:\dev\platform\android\app\src\customized\AndroidManifest.xml:7:5-29:19 Warning:
        application@android:label was tagged at AndroidManifest.xml:7 to replace other declarations but no other declaration present
:app:processCustomizedDebugResources
:app:generateCustomizedDebugSources
:app:incrementalCustomizedDebugJavaCompilationSafeguard
:app:compileCustomizedDebugJavaWithJavac
:app:compileCustomizedDebugJavaWithJavac - is not incremental (e.g. outputs have changed, no previous execution, etc.).
C:\dev\platform\android\app\src\main\java\com\shoutemapp\MainApplication.java:7: error: package com.imagepicker does not exist
import com.imagepicker.ImagePickerPackage;
                      ^
C:\dev\platform\android\app\src\main\java\com\shoutemapp\MainApplication.java:8: error: package com.reactnative.photoview does not exist
import com.reactnative.photoview.PhotoViewPackage;
                                ^
C:\dev\platform\android\app\src\main\java\com\shoutemapp\MainApplication.java:10: error: package com.BV.LinearGradient does not exist
import com.BV.LinearGradient.LinearGradientPackage;
                            ^
C:\dev\platform\android\app\src\main\java\com\shoutemapp\MainApplication.java:50: error: cannot find symbol
            new ImagePickerPackage(),
                ^
  symbol: class ImagePickerPackage
C:\dev\platform\android\app\src\main\java\com\shoutemapp\MainApplication.java:51: error: cannot find symbol
            new PhotoViewPackage(),
                ^
  symbol: class PhotoViewPackage
C:\dev\platform\android\app\src\main\java\com\shoutemapp\MainApplication.java:53: error: cannot find symbol
            new LinearGradientPackage(),
                ^
  symbol: class LinearGradientPackage
Note: C:\dev\platform\android\app\src\main\java\com\shoutemapp\MainApplication.java uses or overrides a deprecated API.
:app:compileCustomizedDebugJavaWithJavac FAILED
Note: Recompile with -Xlint:deprecation for details.
6 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileCustomizedDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```

**Comment by ikatun** [comment]
Also, the iOS does get built successfully, but i get this error from within the app:
http://imgur.com/a/wTg5a

File: scripts/classes/extensions-installer.js [hunk 1]
```diff
@@ -95,13 +102,17 @@ class ExtensionsInstaller {
...
+    _.forEach(excludePackages, packageName => delete packageJsonTemplate.dependencies[packageName]);
     return writePackageJson(packageJsonTemplate)
       .then(() => install())
       .then(() => Promise.all(
         this.localExtensions.map((extension) => installLocalExtension(extension))
       ))
-      .then(() => Promise.resolve(installedExtensions));
+      .then(() =>
+        Promise.resolve(_.filter(installedExtensions, (extension) =>
+          shouldInstallExtension(extension)
```
Comment (zrumenjak) [inline scripts/classes/extensions-installer.js]: Could you do this above, when generating the extensions to install list?

File: scripts/classes/extensions-installer.js [hunk 2]
```diff
@@ -95,13 +102,17 @@ class ExtensionsInstaller {
     const install = this.localExtensions.length ? npmInstall : yarnInstall;
 
     const installedExtensions = [...this.localExtensions, ...this.extensionsToInstall];
-    excludePackages.forEach(packageName => delete packageJsonTemplate.dependencies[packageName]);
+    _.forEach(excludePackages, packageName => delete packageJsonTemplate.dependencies[packageName]);
```
Comment (zrumenjak) [inline scripts/classes/extensions-installer.js]: This function both excludes packages and installs them. Can you extract the logic for excluding packages from this function?

File: android/app/src/main/java/com/shoutemapp/MainApplication.java [hunk 3]
```diff
@@ -40,6 +44,10 @@ public boolean getUseDeveloperSupport() {
     protected List<ReactPackage> getPackages() {
       return Arrays.<ReactPackage>asList(
           new MainReactPackage(),
+            new MapsPackage(),
+            new ImagePickerPackage(),
+            new PhotoViewPackage(),
+            new LinearGradientPackage(),
```
Comment (zrumenjak) [inline android/app/src/main/java/com/shoutemapp/MainApplication.java]: This probably shouldn't be committed. Aren't those dependencies added by `react-native link`?

**Pull request merged**